3_javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,105 @@
+
+
+const user = {
+    userName: "",
+    todos: []
+}
+
+function getReply(command) {
+
+    //command format: Hello my Name is Name
+    if (command.match("Hello my Name is ")) {
+        user.userName = extractName(command)
+        return `Nice to meet you ${user.userName}!`;
+    }
+    //command format: "What is my name?"
+    if (command.match("What is my name?")) {
+        return `Your name is ${user.userName}`;
+    }
+
+    //command format: Add task to my todo
+    const addPrefix = "Add ";
+    const addSuffix = " to my todo";
+    if (command.startsWith(addPrefix) && command.endsWith(addSuffix)) {
+        const task = command.slice(addPrefix.length, -addSuffix.length); // Extract the task dynamically
+        user.todos.push(task);
+        return `${task} added to your todo`;
+    }
+    // remove task to my todo
+    const removePrefix = "Remove ";
+    const removeSuffix = " from my todo";
+    if (command.startsWith(removePrefix) && command.endsWith(removeSuffix)) {
+        const task = command.slice(removePrefix.length, -removeSuffix.length); // Extract the task dynamically  
+        const index = user.todos.indexOf(task);
+        if (index !== -1) {
+            user.todos.splice(index, 1);
+            return `Removed ${task} from your todo`;
+        } else {
+            return `${task} was not found in your todo list`;
+        }
+    }
+    // whats in my todo 
+    if (command.match("What is on my todo?")){
+        const numTodos = user.todos.length;
+        if (numTodos == 0) {
```

Comment:
It is ok to use == in this specific case, but it is worth remembering that there is a difference between == and ===. 
I would say it is safer to usually opt for ===, but it also depends on opinion a bit 😀 
Google around, but f.x. this is quite a good explanation:
https://www.scaler.com/topics/javascript/difference-between-double-equals-and-triple-equals-in-javascript/

Reply:
thanks i will read the article to understand  more he difference